modules/apigee_edge_apidocs/src/Entity/ApiDoc.php

```diff
@@ -122,16 +139,48 @@ public function setCreatedTime(int $timestamp) : ApiDocInterface {
   /**
    * {@inheritdoc}
    */
-  public function isPublished() : bool {
-    return (bool) $this->getEntityKey('status');
+  public function getRevisionUser() {
+    return $this->get('revision_user')->entity;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function setPublished(bool $published) : ApiDocInterface {
-    $this->set('status', $published);
-    return $this;
+  public function preSave(EntityStorageInterface $storage) {
+    parent::preSave($storage);
+
+    // If no revision author has been set explicitly, make the current user
+    // the revision author.
+    if (!$this->getRevisionUser()) {
```

Comment:
I like better what Node does.

```
  // If no revision author has been set explicitly, make the node owner the
    // revision author.
    if (!$this->getRevisionUser()) {
      $this->setRevisionUserId($this->getOwnerId());
    }
```

Reply:
In Node, the owner id is the _uid_ field, which we don't have in the ApiDoc entity. And Node's implementation is a workaround to what we are already doing:
From Node:
```
$fields['uid'] = BaseFieldDefinition::create('entity_reference')
     // ...
      ->setDefaultValueCallback('Drupal\node\Entity\Node::getCurrentUserId')
    // ...
```
and 
```
public static function getCurrentUserId() {
  return [\Drupal::currentUser()->id()];
}
```
So I think the current approach works just as well:
```
if (!$this->getRevisionUser()) {
  $this->setRevisionUserId(\Drupal::currentUser()->id());
}
```

Reply:
> In Node, the owner id is the uid field, which we don't have in the ApiDoc entity.

Should not we have an owner on ApiDoc entities? :) This is kinda an OOTB feature on content entities (nodes) that now we are intentionally removing from API doc entities - although it could be useful in the content moderation workflows, ex.: the owner (creator) of an API documentation cannot publish the API documentation only an other evaluated user.

Reply:
It sounds like a good idea, but not strictly related to this PR. Could we revisit it as another ticket?